x-pack/plugins/observability_solution/apm/common/waterfall/typings.ts

```diff
@@ -7,82 +7,60 @@
 
 export interface WaterfallTransaction {
-  timestamp: TimestampUs;
-  trace: { id: string };
-  service: {
-    name: string;
-    environment?: string;
-  };
-  agent: { name: AgentName };
-  event?: { outcome?: EventOutcome };
-  parent?: { id?: string };
-  processor: { event: 'transaction' };
-  transaction: {
-    duration: { us: number };
-    id: string;
-    name: string;
-    type: string;
-    result?: string;
-  };
-  faas?: {
-    coldstart?: boolean;
-  };
-  span?: {
-    links?: SpanLink[];
-  };
+  'timestamp.us': number[];
+  'trace.id': string[];
+  'service.name': string[];
+  'service.environment'?: string[];
+  'agent.name': AgentName[];
+  'event.outcome'?: EventOutcome[];
+  'parent.id'?: string[];
+  'processor.event': ['transaction'];
+  'transaction.duration.us': number[];
+  'transaction.id': string[];
+  'transaction.name': string[];
+  'transaction.type': string[];
+  'transaction.result'?: string[];
+  'faas.coldstart'?: boolean[];
+  'span.links.span.id'?: string[];
+  'span.links.trace.id'?: string[];
 }
 
 export interface WaterfallSpan {
-  timestamp: TimestampUs;
-  trace: { id: string };
-  service: {
-    name: string;
-    environment?: string;
-  };
-  agent: { name: AgentName };
-  event?: { outcome?: EventOutcome };
-  parent?: { id?: string };
-  processor: { event: 'span' };
-  span: {
-    id: string;
-    type: string;
-    subtype?: string;
-    action?: string;
-    name: string;
-    composite?: {
-      count: number;
-      sum: { us: number };
-      compression_strategy: string;
-    };
-    sync?: boolean;
-    duration: { us: number };
-    links?: SpanLink[];
-  };
-  transaction?: {
-    id: string;
-  };
-  child?: { id: string[] };
+  'timestamp.us': number[];
+  'trace.id': string[];
+  'service.name': string[];
+  'service.environment'?: string[];
+  'agent.name': AgentName[];
+  'event.outcome'?: EventOutcome[];
+  'parent.id'?: string[];
+  'processor.event': ['span'];
+  'span.id': string[];
+  'span.type': string[];
+  'span.subtype'?: string[];
+  'span.action'?: string[];
+  'span.name': string[];
+  'span.composite.count'?: number[];
+  'span.composite.sum.us'?: number[];
+  'span.composite.compression_strategy'?: string[];
+  'span.sync'?: boolean[];
+  'span.duration.us': number[];
+  'span.links.span.id'?: string[];
+  'span.links.trace.id'?: string[];
+  'transaction.id'?: string[];
+  'child.id'?: string[];
 }
 
 export interface WaterfallError {
-  timestamp: TimestampUs;
-  trace?: { id: string };
-  transaction?: { id: string };
-  parent?: { id: string };
-  error: {
-    id: string;
-    log?: {
-      message: string;
-    };
-    exception?: Exception[];
-    grouping_key: string;
-  };
-  service: {
-    name: string;
-  };
+  'timestamp.us': number[];
+  'trace.id'?: string[];
+  'transaction.id'?: string[];
+  'parent.id'?: string[];
+  'error.id': string[];
+  'error.log.message'?: string[];
+  'error.exception'?: Exception[];
```

Comment:
This needs to get resolved

Reply:
It seems like the `error.exception.stacktrace...` fields are not accessible through the `fields` query. Should I try injecting them through the `_sources` response? @felixbarny

Reply:
Hm, yes, I think the exception stack trace (and possibly span links) need to be fetched from _source. Make sure to only fetch the fields you need from source using [source filtering](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/search-fields.html#source-filtering).

Note that the structure will be different for OTel values. So we'll probably want to normalize/convert the data to have a consistent model the UI can work with.